### ISSUE TITLE
Make UpdateControl an enum

### ISSUE THREAD

**Issue by lburgazzoli** [java-operator-sdk/java-operator-sdk#406]
The  method `ResourceController.createOrUpdateResource` should return an instance of `UpdateControl` to instruct the operator about if and what to update. The `UpdateControl` ckass, provides the following helpers:

```java
<T extends CustomResource> UpdateControl<T> updateCustomResource(T cr)
<T extends CustomResource> UpdateControl<T> updateStatusSubResource(T cr)
<T extends CustomResource> UpdateControl<T> updateCustomResourceAndStatus(T cr)
```

Wonder if this could be reduced to an `enum` instead, like:

```Java 
enum UpdateControl {
    UpdateResource,             
    UpdateStatus,
    UpdateResourceAndStatus
}
```

The `ResourceController` signature then would become:

```java
UpdateControl createOrUpdateResource(...)
```

The calling method then would use the custom resource instance which in such case should be the resource received as parameter of the `createOrUpdateResource` method.

Does that make any sense ?

**Comment by metacosm** [comment]
I think it makes sense, indeed.

**Comment by teszes** [comment]
How should we deal with the type parameter though? With an enum, we can't have `UpdateControl<T>` objects.

**Comment by lburgazzoli** [comment]
@teszes I don't think it matters as the ` EventDispatcher<R extends CustomResource>` is already aware of the type parameter and `ResourceController.createOrUpdateResource` is not designed to return a different type

**Comment by lburgazzoli** [comment]
I think we can close this issue
